dpnp/backend/kernels/dpnp_krnl_fft.cpp

```diff
@@ -476,8 +476,12 @@ DPCTLSyclEventRef dpnp_fft_fft_c(DPCTLSyclQueueRef q_ref,
         else if constexpr (std::is_same<_DataType_input, int32_t>::value ||
                            std::is_same<_DataType_input, int64_t>::value)
         {
-            double *array1_copy = reinterpret_cast<double *>(
-                dpnp_memory_alloc_c(q_ref, input_size * sizeof(double)));
+            using CastType = std::conditional_t<
```

Comment:
Need to add a static assertion that sycl::detail::is_complex<_DataType_output> is true.